vllm/entrypoints/openai/serving_pooling.py

```diff
@@ -43,6 +44,9 @@ def _get_data(
         pt_float32 = output.data.to(dtype=torch.float32)
         pooling_bytes = np.array(pt_float32, dtype="float32").tobytes()
         return base64.b64encode(pooling_bytes).decode("utf-8")
+    elif encoding_format == "tensor":
+        tensor_encoding_io = ImageEmbeddingMediaIO()
+        return tensor_encoding_io.encode_base64(output.data)
```

Comment:
I'm not sure I understand this. The pooler output is a torch.Tensor but ImageEmbeddingMediaIO.encode_base64 expects an image. How does this work?

Reply:
I highlighted this in the RFC as part of *Additional features explored to enable tensors:*. 

[ImageMediaIO](https://github.com/mgazz/vllm/blob/tensor_inputs/vllm/multimodal/image.py#L65) expects an image. [ImageEmbeddingMediaIO](https://github.com/mgazz/vllm/blob/tensor_inputs/vllm/multimodal/image.py#L96) expects Tensors. I thought about reusing `ImageEmbeddingMediaIO` to encode tensors output but I had a hard time reconstructing the tensor on the user side when encoded via `encode_base64`. that is why I added [encode_tensor](https://github.com/mgazz/vllm/blob/tensor_inputs/vllm/multimodal/image.py#L101) 

If this makes it too confusing we can have a dedicated class like `TensorIO` that performs the encoding so that we can keep separation of concerns.

Reply:
I opened a separate issue on this: https://github.com/vllm-project/vllm/issues/20427